src/main/java/ui/UpdateProgressWindow.java

```diff
@@ -0,0 +1,144 @@
+package ui;
+
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressBar;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import updater.DownloadProgressTracker;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * UpdateProgressWindow is the class that will display update download progresses
+ */
+public class UpdateProgressWindow {
+    private static final String WINDOW_TITLE = "Update Progress";
+    private Stage window;
+    private VBox windowMainLayout;
+    private final List<DownloadProgressTracker> downloads;
+
+    public UpdateProgressWindow() {
+        this.downloads = new ArrayList<>();
+    }
+
+    /**
+     * shows update download progress window
+     */
+    public void showWindow() {
+        Platform.runLater(() -> {
+                    if (window == null) {
+                        this.createProgressWindow();
+                    } else {
+                        window.show();
+                    }
+
+                    window.toFront();
+                }
+        );
+    }
+
+    /**
+     * hides update download progress window
+     */
+    public void hideWindow() {
+        if (window != null) {
```

Comment:
Supposedly this will never be called when window is not yet created. I will try to initialise the stage on construction.